### ISSUE TITLE
Compiled Bindings: Compilation Error - Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by inforithmics** [xamarin/Xamarin.Forms#4438]
### Description
When using Compiled Bindings on a Type with a Generic Base Class and Binding on A Property with the Generic Type: A Compilation Error occurs with Object reference not set to an instance of an object.
### Steps to Reproduce

1.  Follwing Base Class
```csharp
 public class TestViewModel : TestViewModelBase<string>
    {
        public TestViewModel()
        {
            this.Add("test");
            this.SelectedItem = this.First();
        }        
    }

    public class TestViewModelBase<T> : Collection<string>
    {
        public T SelectedItem { get; set; }
    }
```
2. 
Following xaml with Binding to the Generic Property SelectedItem
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:App14"
             x:Class="App14.MainPage"
             x:DataType="local:TestViewModel">

    <StackLayout>
        <!-- Place new controls here -->        
        <Label Text="Number of Items:"/>
        <Label Text="{Binding SelectedItem}" />        
    </StackLayout>

</ContentPage>
```
3. 

### Expected Behavior
The Binding Works and the Code compiles
### Actual Behavior
The Code doesn't Compile, following error occurs: 
Object reference not set to an instance of an object
### Basic Information

- Version with issue: 3.4.0.1008975
- Last known good version: N/A
- IDE: Visual Studio 2017 15.9.1
- Platform Target Frameworks: 
  - iOS:  12.1 
  - Android:  The version of the Android SDK you are compiling against, e.g. 9.0
  - UWP:  16299 
- Android Support Library Version: v27.0.2.1
- Nuget Packages: N/A
- Affected Devices: N/A

### Reproduction Link

[App14.zip](https://github.com/xamarin/Xamarin.Forms/files/2588736/App14.zip)

### Workaround
Override the Property in the ViewModel than it works.
```csharp
    public class TestViewModel : TestViewModelBase<string>
    {
        public TestViewModel()
        {
            this.Add("test");
            this.SelectedItem = this.First();
        }

        public override string SelectedItem
        {
            get => base.SelectedItem;
            set => this.SelectedItem = value;
        }
    }

    public class TestViewModelBase<T> : Collection<string>
    {
        public virtual T SelectedItem { get; set; }
    }
```

**Comment by StephaneDelcroix** [comment]
I have to check, but I think this was resolved in #4352, and the fix will be released as part of the 3.4.0 release

**Comment by StephaneDelcroix** [comment]
y, I think that is. reopen if it's not fixed in 3.4.0

**Comment by inforithmics** [comment]
Can you reopen this issue because it is not a duplicate of https://github.com/xamarin/Xamarin.Forms/issues/4348 
1. 

I used 3.4.0 Version as stated in Version with issue.

2.       

This binding works now (was bug in issue)
`<Label Text="{Binding Count}" />  `
This binding does not work
`<Label Text="{Binding SelectedItem}" />`
throws Object reference not set to an instance of an object.

The Difference between the two are.

Count is a Property on a Generic Type.
SelectedItem is a Property on Generic Type with a Generic TypeParameter as Type.

So I guess Compiled Bindings tries to find out the Type of this Property and cannot because it is a TypeParameter Type.

Additional Info:

I tried it with the Nightly Nuget package 3.5.1006868 and the issue is still there

**Comment by StephaneDelcroix** [comment]
fixed in #4483

**Comment by inforithmics** [comment]
Thanks a lot for the fix.
